abcclassroom/clone.py

```diff
@@ -63,18 +90,24 @@ def clone_student_repos(args):
 
     if materials_dir is None:
         print(
-            "No course_materials directory set in config.yml. Will clone "
-            "repositories but will not copy any assignment files."
+            "Oops! I couldn't find a course_materials directory location "
+            "in your config.yml file. I will just clone all of the student"
+            "repositories. I can't copy any assignment files to a "
+            "course_materials directory."
         )
+    # This is a really large try block and makes it hard to figure out where
+    # things fail. perhaps break out into a few sub try / excepts?
     try:
         # Create the assignment subdirectory path and ensure it exists
         Path(course_dir, clone_dir, assignment_name).mkdir(exist_ok=True)
         missing = []
+        # If it can't find the roster file you'll get a FileNotFoundError
+        # Also what happens if the roster is not in the correct format??
         with open(roster_filename, newline="") as csvfile:
             reader = csv.DictReader(csvfile)
             for row in reader:
                 student = row["github_username"]
-                # expected columns: identifier,github_username,github_id,name
+                # Expected columns: identifier,github_username,github_id,name
                 repo = "{}-{}".format(assignment_name, student)
                 try:
                     clone_or_update_repo(
```

Comment:
this quietly fails sometimes which is a bit problematic. but i'll deal with that later i think we need to check that the repo exists???

Reply:
We also need to check that authentication is properly setup which is actually why it was failing quietly on the hub. so perhaps that's a separate PR once we get authentication setup for the new github protocols??